source/common/grpc/common.cc

```diff
@@ -42,10 +43,9 @@ Buffer::InstancePtr Common::serializeBody(const google::protobuf::Message& messa
   ASSERT(iovec.len_ >= alloc_size);
   iovec.len_ = alloc_size;
   uint8_t* current = reinterpret_cast<uint8_t*>(iovec.mem_);
-
-  // TODO(fengli79): Make this less ugly.
   *current++ = 0; // flags
-  *reinterpret_cast<uint32_t*>(current) = htonl(size);
+  uint32_t nsize = htonl(size);
```

Comment:
You should be able to reinterpret_cast<const void*> and the second argument of memcpy is const void*